tests/e2e/flowtest/test_selinux_relabeling_solution.py

```diff
@@ -0,0 +1,255 @@
+            namespace=config.ENV_DATA["cluster_namespace"],
+        )
+
+        tmploc = ntar_loc.split("/")[-1]
+        ocp_obj.exec_oc_cmd(
+            f"rsync {ntar_loc} {pod_name}:{constants.FLEXY_MNT_CONTAINER_DIR}",
+            timeout=300,
+        )
+        ocp_obj.exec_oc_cmd(
+            f"exec {pod_name} -- mkdir {constants.FLEXY_MNT_CONTAINER_DIR}/x"
+        )
+        ocp_obj.exec_oc_cmd(
+            f"exec {pod_name} -- /bin/tar xf"
+            f" {constants.FLEXY_MNT_CONTAINER_DIR}/{tmploc}/{TARFILE}"
+            f" -C {constants.FLEXY_MNT_CONTAINER_DIR}/x",
+            timeout=3600,
+        )
+        log.info("cephfs test files created on pod")
+
+    def calculate_md5sum(pod_obj):
+        pod_name = pod_obj.name
+        data_path = f'"{constants.FLEXY_MNT_CONTAINER_DIR}/x"'
+        ocp_obj = ocp.OCP(
+            kind=constants.POD,
+            namespace=config.ENV_DATA["cluster_namespace"],
+        )
+        random_file = ocp_obj.exec_oc_cmd(
+            f"exec -it {pod_name} -- /bin/bash"
+            f' -c "find {data_path} -type f | "shuf" -n 10"',
+            timeout=300,
+        )
+
+        md5sum_pod_data = pod.cal_md5sum(pod_obj=pod_obj, file_name=random_file)
+        return md5sum_pod_data
+
+    def data_integrity_check(self):
+        pod_objs = pod.get_all_pods(
+            namespace=self.namespace, selector=["nfs-test-pod-2"], selector_label="name"
+        )
+        pod_obj = pod_objs[0]
+        log.info(f"pod obj name----{pod_obj.name}")
+        pod_name = pod_obj.name
+        ocp_obj = ocp.OCP(
+            kind=constants.POD,
+            namespace=config.ENV_DATA["cluster_namespace"],
+        )
+
+        data_path = f'"{constants.FLEXY_MNT_CONTAINER_DIR}/x"'
+        random_file = ocp_obj.exec_oc_cmd(
+            f"exec -it {pod_name} -- /bin/bash"
+            f' -c "find {data_path} -type f | "shuf" -n 1"',
+            timeout=300,
+        )
+
+        ini_md5sum_pod_data = pod.cal_md5sum(pod_obj=pod_obj, file_name=random_file)
+        start_time = time.time()
+        pod_obj.delete(wait=False)
+        assert wait_for_pods_to_be_running(timeout=600, sleep=15)
+        end_time = time.time()
+        total_time = end_time - start_time
+        log.info(f"Time taken by pod to restart is  {total_time}")
+
+        pod_objs = pod.get_all_pods(
+            namespace=self.namespace, selector=["nfs-test-pod-2"], selector_label="name"
+        )
+        pod_obj = pod_objs[0]
+        fin_md5sum_pod_data = pod.cal_md5sum(pod_obj=pod_obj, file_name=random_file)
+        assert ini_md5sum_pod_data == fin_md5sum_pod_data
+        return total_time
+
+    def cleanup(self):
+        # Delete deployment config
+        cmd_delete_deployment_config = "delete dc nfs-test-pod"
+        self.storage_cluster_obj.exec_oc_cmd(cmd_delete_deployment_config)
+        # Delete pvc
+        self.nfs_pvc_obj.delete()
+        log.info("Teardown complete")
+
+    @ignore_leftovers
```

Comment:
what will be leftovers here? 
May it be hypothetically that resources may create after this test? if so it will fail the next test which is not labeled with @ignore_leftovers deco